app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -334,6 +341,9 @@
     private PopupMenu playbackSpeedPopupMenu;
     private PopupMenu captionPopupMenu;
 
+    private boolean isDefaultResolutionOverrideOrMaxVideoSizeAndFrameRateSet = false;
```

Comment:
Also here, maybe use `isUsingDefaultResolutionForLivestreams` which is shorter and more meaningful. I can't understand what `isDefaultResolutionOverrideOrMaxVideoSizeAndFrameRateSet` means, I just see that it is only used inside `setDefaultResolutionForLivestreams` and it prevents that function from being executed multiple times

Reply:
Changed to `isDefaultResolutionSetForLivestreams`.